R/JHU.R

```diff
@@ -0,0 +1,162 @@
+    #' @importFrom tidyr pivot_longer pivot_wider replace_na
+    #' @importFrom rlang .data
+    clean_common = function() {
+      self$data$clean <- lapply(self$data$raw,
+        pivot_longer,
+        cols = 5:last_col(),
+        names_to = "Date",
+        values_to = "value"
+      )
+      self$data$clean <- self$data$clean %>%
+        bind_rows(.id = "variable") %>%
+        pivot_wider(names_from = variable, values_from = value) %>%
+        select(
+          .data$Date,
+          .data$`Province/State`,
+          .data$`Country/Region`,
+          .data$daily_confirmed,
+          .data$daily_deaths,
+          .data$daily_recovered
+        ) %>%
+        mutate(
+          Date = lubridate::mdy(.data$Date),
+          daily_confirmed = as.numeric(.data$daily_confirmed),
+          daily_deaths = as.numeric(.data$daily_deaths),
+          daily_recovered = as.numeric(.data$daily_recovered),
+          level_2_region_code = NA_real_
+        ) %>%
+        rename(
+          date = .data$Date,
+          level_2_region = .data$`Province/State`,
+          level_1_region = .data$`Country/Region`,
+          cases_total = .data$daily_confirmed,
+          deaths_total = .data$daily_deaths,
+          recovered_total = .data$daily_recovered
+        ) %>%
+        replace_na(
+          list(
+            level_1_region = "Unknown",
+            level_2_region = "Unknown"
+          )
+        ) %>%
+        full_join(
+          self$codes_lookup$`1`,
+          by = c("level_1_region" = "Country_Region")
+        )
+    },
+
+    #' @description JHU specific country level data cleaning
+    #' @importFrom dplyr select summarise group_by
+    #' @importFrom rlang .data
+    clean_level_1 = function() {
+      self$data$clean <- self$data$clean %>%
+        select(
+          .data$date, .data$level_1_region_code, .data$level_1_region,
+          .data$cases_total, .data$deaths_total, .data$recovered_total
+        ) %>%
+        group_by(
+          .data$date, .data$level_1_region_code,
+          .data$level_1_region
+        ) %>%
+        summarise(across(where(is.double), sum))
+    },
+
+    #' @description JHU specific state level data cleaning
+    #' @importFrom dplyr select
+    #' @importFrom rlang .data
+    clean_level_2 = function() {
+      self$data$clean <- self$data$clean %>%
+        select(
+          .data$date, .data$level_1_region_code, .data$level_1_region,
+          .data$level_2_region_code, .data$level_2_region,
+          .data$cases_total, deaths_total, .data$recovered_total
+        )
+    },
+
+    #' @description Specific return settings for the JHU dataset.
+    #' @importFrom dplyr group_by ungroup select arrange
+    #' @importFrom tidyr fill
+    return = function() {
+      self$data$return <- self$data$processed
+      if (!self$totals) {
+        if (self$level == "1") {
+          self$data$return <- self$data$return %>%
+            select(
```

Comment:
I can't remember why we need this custom return? No problem with it being here in general.

Reply:
Yeah it was in the ECCD and WHO stuff so thought I'd add it here, not really sure what it adds but keeps it consitent.

Reply:
Fair enough. Can review at a later date!